### ISSUE TITLE
gopts: No error on missing argument 

### ISSUE THREAD

**Issue by qwepoizt** [ElektraInitiative/libelektra#4042]
Forked from #4029.

When using the HL API and the gopts plugin, CLI options that have `opt/arg=required` and are passed without a value are not validated.

E.g. `/myapp --mb` where the option `--mb` requires an argument, generates no error.

## Steps to Reproduce the Problem
0. Execute the steps below which follow the tutorial [High-level API (with code-generation)](https://www.libelektra.org/tutorials/high-level-api). 

1. Create file test.ini with content:
```
[]
mountpoint = test.ecf

[myboolean]
type = boolean
default = 1
example = 0
opt/long = mb
opt/arg = required
```

2. Execute `kdb gen -F ni=test.ini highlevel "/sw/example/myapp/#0/current" conf`

3. Create file myapp.c with content:
```c#include <stdlib.h>
#include <stdio.h>
#include <elektra.h>
#include "conf.h"

static void onFatalError(ElektraError * error) {
    fprintf(stderr, "An error occured during an Elektra operation: %s \n", elektraErrorDescription(error));
    elektraErrorReset(&error);
    exit(EXIT_FAILURE);
}

int main(int argc, const char * const * argv, const char * const * envp) {
    exitForSpecload (argc, argv);
    ElektraError * error = NULL;
    Elektra * elektra = NULL;
    int rc = loadConfiguration (&elektra, argc, argv, envp, &error);

    if (rc == -1)
    {
        fprintf (stderr, "An error occurred while opening Elektra: %s", elektraErrorDescription (error));
        elektraErrorReset (&error);
        exit (EXIT_FAILURE);
    }

    if (rc == 1)
    {
        // help mode - application was called with '-h' or '--help'
        // for more information see "Command line options" below
        printHelpMessage (elektra, NULL, NULL);
        elektraClose (elektra);
        exit (EXIT_SUCCESS);
    }

    elektraFatalErrorHandler (elektra, onFatalError);


    fprintf(stdout, "myboolean is: %d \n", elektraGetMyboolean(elektra));
    fprintf(stdout, "elektraGet...() was executed for all keys available in specification!\n");
}
```

4. Link and compile

`` cc -g -O0 myapp.c conf.c `pkg-config --cflags --libs elektra-codegen` -I. -o myapp -Wl,-rpath `pkg-config --variable=libdir elektra-codegen` ``

5. Mount specification file and spec-mount via generated `conf.mount.sh`:
``$  APP_PATH=`pwd`/myapp sh conf.mount.sh``

6. Execute ` ./myapp --mb` (i.e. passing no argument for `--mb`)

**Expected**:  
`loadConfiguration ()` fails with error message "option --mb requires an argument!"

**Actual**:  
Missing value ignored, default value applied.
```/myapp --mb
myboolean is: 1 
elektraGet...() was executed for all keys available in specification!
```

## System Information
- Elektra Version: 0.9.7
- Operating System: Ubuntu 20.04
- Versions of other relevant software: n/a

**Comment by qwepoizt** [comment]
(mentioning @markus2330 as discussed.)

**Comment by qwepoizt** [comment]
The gopts plugin actually recognizes this as an error, but the error is swallowed within `kdb.c`. See #4044

**Comment by markus2330** [comment]
Can you please update the top-post to summarize what this issue is about?

**Comment by qwepoizt** [comment]
> Can you please update the top-post to summarize what this issue is about?

Yes, I did!

**Comment by qwepoizt** [comment]
This was fixed via #4045.

Actual result is now as expected:

```
./myapp --mb
An error occurred while opening Elektra: Validation Semantic: Missing argument for long option: --mb 
```

**Comment by markus2330** [comment]
Thank you for fixing the issue! :1st_place_medal:
